src/ucs/memory/rcache.c

```diff
@@ -999,6 +1000,14 @@ static void ucs_rcache_global_list_remove(ucs_rcache_t *rcache) {
     pthread_mutex_unlock(&ucs_rcache_global_list_lock);
 }
 
+static void
+ucs_rcache_vfs_show_num_regions(void *obj, ucs_string_buffer_t *strb)
+{
+    ucs_rcache_t *rcache = obj;
+
+    ucs_string_buffer_appendf(strb, "%lu\n", rcache->num_regions);
+}
```

Comment:
1. IMO we should add generic function ucs_vfs_show_ulong with custom argument (pointer to the ulong value)
2. print also total size, max regions, max size
3. print length of inv_q and gc_list

Reply:
The function is called from vfs. Do you suggest to store the pointer to the value inside the vfs node?

E.g.:
```C
static void
ucs_vfs_read_ro_file(ucs_vfs_node_t *node, ucs_string_buffer_t *strb)
{
    ucs_vfs_node_t *parent_node = node->parent;

    ucs_assert(ucs_vfs_check_node(node, UCS_VFS_NODE_TYPE_RO_FILE) == 1);

    while (ucs_vfs_check_node(parent_node, UCS_VFS_NODE_TYPE_SUBDIR)) {
        parent_node = parent_node->parent;
    }

    ucs_spin_unlock(&ucs_vfs_obj_context.lock);

- node->text_cb(parent_node->obj, strb);
+ node->text_cb.cb(parent_node->text_cb.value, strb);

    ucs_spin_lock(&ucs_vfs_obj_context.lock);
}
```

Reply:
We can use node::obj to store the pointer to the value, but this doesn't look obvious, IMO.

Reply:
IMO, like this:
```c
node->text_cb(parent_node->obj, node->arg, strb);
```
then:
```c
static void
ucs_vfs_show_ulong(void *obj, void *arg, ucs_string_buffer_t *strb)
{
   ucs_string_buffer_appendf(strb, "%lu\n", *(unsigned long*)arg);
}

ucs_vfs_obj_add_ro_file(self, ucs_vfs_show_ulong, &rcache->num_regions, "num_regions");
ucs_vfs_obj_add_ro_file(self, ucs_vfs_show_ulong, &rcache->total_size, "total_size");

```

Reply:
got it!